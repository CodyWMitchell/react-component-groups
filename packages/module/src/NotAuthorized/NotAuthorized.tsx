import React from 'react';
import { Button, EmptyState, EmptyStateBody, EmptyStateIcon, EmptyStateProps, EmptyStateVariant, Title } from '@patternfly/react-core';
import { LockIcon } from '@patternfly/react-icons';
import { createUseStyles } from 'react-jss';

export interface NotAuthorizedProps extends Omit<EmptyStateProps, 'children' | 'title'> {
  /** Service name displayed in the title */
  serviceName?: string;
  /** Icon displayed above the title */
  icon?: React.ComponentType;
  /** Custom description */
  description?: React.ReactNode;
  /** Shows link to the previous page */
  showReturnButton?: boolean;
  /** Custom previous page button text */
  prevPageButtonText?: React.ReactNode;
  /** Custom className */
  className?: string;
  /** Custom title */
  title?: React.ReactNode;
  /** Custom state actions */
  actions?: React.ReactNode;
  /** Custom landing page button text */
  toLandingPageText?: React.ReactNode;
  /** Custom landing page button URL */
  toLandingPageUrl?: string;
}

const useStyles = createUseStyles({
  title: {
    maxWidth: '540px',
    marginLeft: 'auto',
    marginRight: 'auto',
  },
})

export const NotAuthorized: React.FunctionComponent<NotAuthorizedProps> = ({
  prevPageButtonText = 'Return to previous page',
  toLandingPageText = 'Go to landing page',
  toLandingPageUrl = ".",
  actions = null,
  serviceName,
  title = `You do not have access to ${serviceName}`,
  icon: Icon = LockIcon,
  description = 'Contact your system administrator(s) for more information.',
  showReturnButton = true,
  className,
  ...props
}: NotAuthorizedProps) => {
  const classes = useStyles();

  return (
    <EmptyState variant={EmptyStateVariant.full} className={className} {...props}>
      <EmptyStateIcon icon={Icon} />
      <Title className={classes.title} headingLevel="h5" size="lg">
        {title}
      </Title>
      <EmptyStateBody>{description}</EmptyStateBody>
      {actions}
      {showReturnButton &&
        (document.referrer ? (
          <Button variant="primary" onClick={() => history.back()}>
            {prevPageButtonText}
          </Button>
        ) : (
          <Button variant="primary" component="a" href={toLandingPageUrl}>
            {toLandingPageText}
          </Button>
        ))}
    </EmptyState>
  );
};

export default NotAuthorized;
