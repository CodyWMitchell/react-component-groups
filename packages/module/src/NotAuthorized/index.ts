export { default } from './NotAuthorized'
export * from './NotAuthorized'
