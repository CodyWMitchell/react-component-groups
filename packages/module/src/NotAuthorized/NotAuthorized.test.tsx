import React from 'react';
import { render } from '@testing-library/react';
import { QuestionCircleIcon } from '@patternfly/react-icons';
import NotAuthorized from './NotAuthorized';

describe('NotAuthorized component', () => {
  const initialProps = {
    serviceName: 'Foo',
  };
  it('should render', () => {
    const { container } = render(<NotAuthorized {...initialProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should apply custom styles', () => {
    const { container } = render(<NotAuthorized {...initialProps} className="something" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should use custom icon', () => {
    const { container } = render(<NotAuthorized {...initialProps} icon={QuestionCircleIcon} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not show buttons', () => {
    const { container } = render(<NotAuthorized {...initialProps} showReturnButton={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should show custom description', () => {
    const { container } = render(<NotAuthorized {...initialProps} description="Some text" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should show custom title', () => {
    const { container } = render(<NotAuthorized title="Custom title" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should show custom actions', () => {
    const actions = [
      <button id="action-one" key="1">
        1
      </button>,
      <button id="action-one" key="2">
        2
      </button>,
    ];
    const { container } = render(<NotAuthorized {...initialProps} actions={actions} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});