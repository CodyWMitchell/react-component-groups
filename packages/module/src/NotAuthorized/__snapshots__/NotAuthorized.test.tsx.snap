// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotAuthorized component should apply custom styles 1`] = `
<div
  class="pf-c-empty-state something"
>
  <div
    class="pf-c-empty-state__content"
  >
    <svg
      aria-hidden="true"
      class="pf-c-empty-state__icon"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
      />
    </svg>
    <h5
      class="pf-c-title pf-m-lg title-0-2-1"
      data-ouia-component-id="OUIA-Generated-Title-2"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe="true"
    >
      You do not have access to Foo
    </h5>
    <div
      class="pf-c-empty-state__body"
    >
      Contact your system administrator(s) for more information.
    </div>
    <a
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-2"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      href="."
    >
      Go to landing page
    </a>
  </div>
</div>
`;

exports[`NotAuthorized component should not show buttons 1`] = `
<div
  class="pf-c-empty-state"
>
  <div
    class="pf-c-empty-state__content"
  >
    <svg
      aria-hidden="true"
      class="pf-c-empty-state__icon"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
      />
    </svg>
    <h5
      class="pf-c-title pf-m-lg title-0-2-1"
      data-ouia-component-id="OUIA-Generated-Title-4"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe="true"
    >
      You do not have access to Foo
    </h5>
    <div
      class="pf-c-empty-state__body"
    >
      Contact your system administrator(s) for more information.
    </div>
  </div>
</div>
`;

exports[`NotAuthorized component should render 1`] = `
<div
  class="pf-c-empty-state"
>
  <div
    class="pf-c-empty-state__content"
  >
    <svg
      aria-hidden="true"
      class="pf-c-empty-state__icon"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
      />
    </svg>
    <h5
      class="pf-c-title pf-m-lg title-0-2-1"
      data-ouia-component-id="OUIA-Generated-Title-1"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe="true"
    >
      You do not have access to Foo
    </h5>
    <div
      class="pf-c-empty-state__body"
    >
      Contact your system administrator(s) for more information.
    </div>
    <a
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-1"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      href="."
    >
      Go to landing page
    </a>
  </div>
</div>
`;

exports[`NotAuthorized component should show custom actions 1`] = `
<div
  class="pf-c-empty-state"
>
  <div
    class="pf-c-empty-state__content"
  >
    <svg
      aria-hidden="true"
      class="pf-c-empty-state__icon"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
      />
    </svg>
    <h5
      class="pf-c-title pf-m-lg title-0-2-1"
      data-ouia-component-id="OUIA-Generated-Title-7"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe="true"
    >
      You do not have access to Foo
    </h5>
    <div
      class="pf-c-empty-state__body"
    >
      Contact your system administrator(s) for more information.
    </div>
    <button
      id="action-one"
    >
      1
    </button>
    <button
      id="action-one"
    >
      2
    </button>
    <a
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-6"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      href="."
    >
      Go to landing page
    </a>
  </div>
</div>
`;

exports[`NotAuthorized component should show custom description 1`] = `
<div
  class="pf-c-empty-state"
>
  <div
    class="pf-c-empty-state__content"
  >
    <svg
      aria-hidden="true"
      class="pf-c-empty-state__icon"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
      />
    </svg>
    <h5
      class="pf-c-title pf-m-lg title-0-2-1"
      data-ouia-component-id="OUIA-Generated-Title-5"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe="true"
    >
      You do not have access to Foo
    </h5>
    <div
      class="pf-c-empty-state__body"
    >
      Some text
    </div>
    <a
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-4"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      href="."
    >
      Go to landing page
    </a>
  </div>
</div>
`;

exports[`NotAuthorized component should show custom title 1`] = `
<div
  class="pf-c-empty-state"
>
  <div
    class="pf-c-empty-state__content"
  >
    <svg
      aria-hidden="true"
      class="pf-c-empty-state__icon"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
      />
    </svg>
    <h5
      class="pf-c-title pf-m-lg title-0-2-1"
      data-ouia-component-id="OUIA-Generated-Title-6"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe="true"
    >
      Custom title
    </h5>
    <div
      class="pf-c-empty-state__body"
    >
      Contact your system administrator(s) for more information.
    </div>
    <a
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-5"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      href="."
    >
      Go to landing page
    </a>
  </div>
</div>
`;

exports[`NotAuthorized component should use custom icon 1`] = `
<div
  class="pf-c-empty-state"
>
  <div
    class="pf-c-empty-state__content"
  >
    <svg
      aria-hidden="true"
      class="pf-c-empty-state__icon"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
      />
    </svg>
    <h5
      class="pf-c-title pf-m-lg title-0-2-1"
      data-ouia-component-id="OUIA-Generated-Title-3"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe="true"
    >
      You do not have access to Foo
    </h5>
    <div
      class="pf-c-empty-state__body"
    >
      Contact your system administrator(s) for more information.
    </div>
    <a
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-3"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      href="."
    >
      Go to landing page
    </a>
  </div>
</div>
`;
